rust/src/editor.rs

```diff
@@ -94,8 +95,8 @@ impl Editor {
         self.add_delta(sel_interval, Rope::from(s), new_cursor);
     }
 
-    fn set_cursor_impl(&mut self, offset: usize, set_start: bool, hard: bool) {
```

Comment:
I'm not sure whether I prefer this, though it does work. I was thinking of keeping this.